kani-compiler/src/codegen_cprover_gotoc/codegen/statement.rs

```diff
@@ -76,17 +76,50 @@ impl<'tcx> GotocCtx<'tcx> {
                 self.codegen_set_discriminant(dest_ty, dest_expr, *variant_index, location)
             }
             StatementKind::StorageLive(var_id) => {
-                if self.queries.args().ignore_storage_markers {
+                if !self.current_fn().is_address_taken_local(*var_id) {
                     Stmt::skip(location)
                 } else {
-                    Stmt::decl(self.codegen_local(*var_id), None, location)
+                    let global_dead_object = self.ensure_global_var(
```

Comment:
Can you please encapsulate this declaration inside the goto bindings create?

Reply:
Addressed in https://github.com/model-checking/kani/pull/2995/commits/7f996e70d98cd9a7e9f5ff985688599e45673898.